components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -421,4 +431,54 @@ public static boolean isUserPortalURL(org.wso2.carbon.identity.recovery.model.Pr
         return false;
     }
 
+    /**
+     * Check if the exception contains a password pattern violation message and act accordingly
+     *
+     * @param exception An UserStoreException
+     * @throws IdentityRecoveryClientException If exception's message contains a password pattern violation message
+     */
+    public static void checkPasswordPatternViolation(UserStoreException exception, int tenantId)
+            throws IdentityRecoveryClientException {
+
+        if (StringUtils.isBlank(exception.getMessage())) {
+            return;
+        }
+
+        String exceptionMessage = exception.getMessage();
+        if (((StringUtils.indexOfAny(exceptionMessage, pwdPatternViolations) >= 0) && StringUtils
+                .containsIgnoreCase(exceptionMessage, PASSWORD_LENGTH_TOKEN)) || exceptionMessage
+                .contains(UserCoreErrorConstants.ErrorMessages.ERROR_CODE_INVALID_PASSWORD.getCode())) {
+
+            UserStoreManager userStoreManager = getUserStoreManager(tenantId);
```

Comment:
load userStoreManager for tenant and as well as userstore domain

Reply:
Fixed